Question5.java

```diff
@@ -0,0 +1,152 @@
+//Stack Interface
+interface Stack<Type>
+{
+    void push(Type data) throws StackSizeExceeded;
+    void pop();
+    void peek();
+    int size();
+}
+
+//Exception Classes
+class StackSizeExceeded extends Exception
+{
+    public StackSizeExceeded(String message) {super(message);}
+}
+class IllegalArgumentException extends Exception
+{
+    public IllegalArgumentException(String message) {super(message);}
+}
+
+//Limited Stack Class
+class LimitedStack<Type> implements Stack<Type>
+{
+    Node head;
+    int max_stack;
+
+    public LimitedStack(int max) throws IllegalArgumentException //LimitedStack Constructor
+    {
+        if(max <= 0) //Validate Argument
+        {
+            throw new IllegalArgumentException("Illegal Argument");
+        }
+        else max_stack=max;
+    }
+
+    private class Node //Create Node
+    {
+        Type data;
+        Node next;
+    }
+
+    @Override
+    public void push(Type data) throws StackSizeExceeded //Push method
+    {
+        if (size()<max_stack)
+        {
+            Node node = new Node();
+            node.data = data;
+            node.next = null;
+
+            if(head==null) head = node;
+
+            else
+            {
+                node.next=head;
+                head = node;
+            }
+        }
+        else throw new StackSizeExceeded("Stack size exceeded");
+
+
+    }
+
+    @Override
+    public void pop() //Pop method
+    {
+        if (size()!=0)
+        {
+            System.out.println(head.data);
+            head=head.next;
+        }
+    }
+
+    @Override
+    public void peek() //Peek method
+    {
+        if (size()!=0)
+        {
+            System.out.println(head.data);
+        }
+    }
+
+    @Override
+    public int size() //Size method
```

Comment:
Can we keep a counter and update during push/pop to track the size instead of iterating the entire loop everytime?

Reply:
This issue has been resolved.